src/firefly/js/ui/ImageSearchPanelV2.jsx

```diff
@@ -0,0 +1,432 @@
+            if (wprs.length === 1) {
+                wpSet.push(wprs[0]);
+            } else { wpSet.push(undefined); }
+        });
+        dispatchPlotImage({threeColor:true, wpRequest: wpSet});
+
+    } else {
+        const wprs = makeWebPlotRequests(get(allFields, FG_KEYS.single), imageMasterData);
+        wprs.forEach( (r) => dispatchPlotImage({wpRequest:r}));
+    }
+}
+
+var outputMessage = (errMsg) => errMsg&&showInfoPopup(errMsg, 'Load Selected Image Error');
+function getAllGroupFields(...keys) {
+    return keys.reduce((prev, fg) => (Object.assign(prev, {[fg]: FieldGroupUtils.getGroupFields(fg)})), {});
+}
+
+
+/**
+ * Creates one or more WebPlotRequest from the given parameters
+ * @param request   the request parameters from the input fields
+ * @param imageMasterData   the full imageMasterData when IRSA is selected
+ * @returns {WebPlotRequest[]}
+ */
+function makeWebPlotRequests(request, imageMasterData){
+
+    if (!request || request.imageSource === 'none') {
+        return [];
+    } else if (request.imageSource === 'upload') {
+        return [makeFitsWebRequest(request)];
+
+    } else if (request.imageSource === 'url') {
+        return [makeURLWebRequest(request)];
+    } else {
+        const wp = parseWorldPt(request[ServerParams.USER_TARGET_WORLD_PT]);
+        const radius= request.conesize;
+
+        const imageIdList= [];
+        Object.keys(request).filter( (k) => k.startsWith('IMAGES_')).forEach( (p) => {
+            const list= request[p].split(',');
+            list.forEach( (e) => imageIdList.push(e));
+        });
+        const paramAry= imageMasterData.filter( (d) => imageIdList.includes(d.imageId));
+        console.log(paramAry);
+        return paramAry.map( (d) => makeWPRequest(wp, radius, d.plotRequestParams));
+    }
+}
+
+
+//=========utility?====================
+
+const rootId='conceptPid-';
+let rootIdCnt=1;
+function makeWPRequest(wp, radius, params) {
+    const inReq= Object.assign( {
+        [WPConst.PLOT_ID] : rootId+rootIdCnt,
+        [WPConst.WORLD_PT] : wp.toString(),
+        [WPConst.SIZE_IN_DEG] : radius+'',
+        [WPConst.PLOT_GROUP_ID] : 'multiImageGroup',
+        [WPConst.GROUP_LOCKED] : 'true',
+    }, params);
+
+    rootIdCnt++;
+    return WebPlotRequest.makeFromObj(inReq);
+}
+
+function makeURLWebRequest(request) {
+
+    var url = get(request, 'txURL');
+    const params = {
+        [WPConst.PLOT_ID] : rootId+rootIdCnt,
+        [WPConst.URLKEY] : url,
+        [WPConst.PLOT_GROUP_ID] : 'multiImageGroup',
+        [WPConst.GROUP_LOCKED] : 'true'
+    };
+    rootIdCnt++;
+
+    return  WebPlotRequest.makeFromObj(params);
+}
+
+function makeFitsWebRequest(request, color='') {
+
+    var fits = get(request, 'fileUpload');
+
+    const params = {
+        [WPConst.PLOT_ID] : rootId+rootIdCnt,
+        [WPConst.FILE] : fits,
+        [WPConst.PLOT_GROUP_ID] : 'multiImageGroup',
+        [WPConst.GROUP_LOCKED] : 'true'
```

Comment:
`RequestType` is normally implied but when you add workspace you will have to set it.
`[WPConst.TYPE] : RequestType.WORKSPACE`